lib/chef/resource/windows_package.rb

```diff
@@ -33,7 +33,9 @@ class WindowsPackage < Chef::Resource::Package
 
       def initialize(name, run_context=nil)
         super
-        @source ||= source(@package_name)
+        if uri_scheme?(@package_name) || ::File.exists?(canonical_path(@package_name))
```

Comment:
Something seems wrong here. What if I have a recipe that does a `cookbook_file` to for getting the package, and then a `windows_package`? I think the file will be neither a uri or exist, and `@source` will not be set?